list/list.go

```diff
@@ -0,0 +1,141 @@
+package list
+
+import (
+	"fmt"
+	"reflect"
+)
+
+// List holds a collection of items that can be displayed with an N number of
+// visible items. The list can be moved up, down by one item of time or an
+// entire page (ie: visible size). It keeps track of the current selected item.
+type List struct {
+	items  []interface{}
+	cursor int // cursor holds the index of the current selected item
+	size   int // size is the number of visible options
+	start  int
+}
+
+// New creates and initializes a list. Items must be a slice type and size must
+// be greater than 0.
+func New(items interface{}, size int) (*List, error) {
+	if size < 1 {
+		return nil, fmt.Errorf("list size %d must be greater than 0", size)
+	}
+
+	if items == nil || reflect.TypeOf(items).Kind() != reflect.Slice {
+		return nil, fmt.Errorf("items %v is not a slice", items)
+	}
+
+	l := &List{size: size}
+
+	slice := reflect.ValueOf(items)
```

Comment:
Might be good to make sure the `items` var is a slice before you treat it like a slice if things might panic with `slice.Kind() == reflect.Slice`.

https://godoc.org/reflect#Value.Kind

Reply:
I should finish my coffee before I do reviews...